core/server/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -268,7 +268,14 @@ public void accessBlock(long sessionId, long blockId) throws BlockDoesNotExistEx
   public void commitBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, BlockDoesNotExistException, InvalidWorkerStateException,
       IOException, WorkerOutOfSpaceException {
-    mBlockStore.commitBlock(sessionId, blockId);
+    // NOTE: this may be invoked multiple times due to retry on client side.
+    // TODO(binfan): find a better way to handle retry logic
+    try {
+      mBlockStore.commitBlock(sessionId, blockId);
+    } catch (BlockAlreadyExistsException e) {
+      LOG.debug("Block {} has been in block store, this could be a retry due to master-side RPC "
```

Comment:
The master side operation should be okay, but we will need to check if the block already is accounted for before update the worker's bytes used/free.